src/Symfony/Component/Cache/Traits/AbstractTrait.php

```diff
@@ -255,7 +255,8 @@ private function getId($key)
             return $this->namespace.$this->namespaceVersion.$key;
         }
         if (\strlen($id = $this->namespace.$this->namespaceVersion.$key) > $this->maxIdLength) {
-            $id = $this->namespace.$this->namespaceVersion.substr_replace(base64_encode(hash('sha256', $key, true)), ':', -22);
+            $adjustedKey = substr_replace(base64_encode(hash('sha256', $key, true)), ':', -(strlen($this->namespaceVersion) + 22));
```

Comment:
No need for the line split, we're ok with longer lines. Could you add a `\` just before strlen also please?